src/main/java/com/adobe/aem/compgenerator/utils/HTMLUtils.java

```diff
@@ -0,0 +1,103 @@
+
+import com.adobe.aem.compgenerator.Constants;
+import com.adobe.aem.compgenerator.models.GenerationConfig;
+import com.adobe.aem.compgenerator.models.Property;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class HTMLUtils {
+
+    public static String renderHtml(GenerationConfig generationConfig) {
+        List<Property> props = getAllProperties(generationConfig);
+        String slingModel = StringUtils.uncapitalize(generationConfig.getJavaFormatedName()) + "Model";
+        StringBuilder renderedHtml = new StringBuilder();
+
+        for (Property prop : props) {
+            String label = "label_not_defined";
+            if (prop.getLabel() != null) {
+                label = prop.getLabel();
+            }
+
+            if (Constants.TYPE_CHECKBOX.equals(prop.getType())) {
+                renderedHtml.append(generateParagraphHtml(label,
+                        prop.getField(),
+                        slingModel,
+                        " ? 'checked' : 'unchecked'"
+                ));
+            } else if (Constants.TYPE_DATEPICKER.equals(prop.getType())) {
+                renderedHtml.append(generateParagraphHtml(label,
+                        prop.getField(),
+                        slingModel,
+                        ".time.toGMTString"
+                ));
+            } else if (Constants.TYPE_IMAGE.equals(prop.getType())) {
+                renderedHtml.append(generateImageHtml(label, prop.getField(), slingModel));
+            } else if (Constants.TYPE_MULTIFIELD.equals(prop.getType())) {
+                renderedHtml.append(generateListHtml(prop, slingModel));
+            } else {
+                renderedHtml.append(generateParagraphHtml(label,
+                        prop.getField(),
+                        slingModel,
+                        StringUtils.EMPTY));
+            }
+        }
+        return renderedHtml.toString();
+    }
+
+    private static String generateImageHtml(String label, String field, String slingModel) {
+        return "<p>" +
+                label +
+                ": <img src=\"" +
+                slingModel +
+                "." +
+                field +
+                ".src}\"/></p>\n";
+    }
+
+    private static String generateListHtml(Property prop, String slingModel) {
+        String initialListHtml = "<div data-sly-list=\"${" +
+                slingModel +
+                "." +
+                prop.getField() +
+                "}\">\n\t<p>";
+        if (prop.getItems().size() > 1) {
+            StringBuilder items = new StringBuilder(initialListHtml);
+            for (Property property : prop.getItems()) {
+                items.append(property.getLabel())
+                        .append(": ${item.")
+                        .append(property.getField())
+                        .append("}\n\t");
```

Comment:
I see a couple `\t` in the HTML generated code - any reason for those tabs?

Reply:
Hm, where exactly, I didn't get any. I used them in java to get nice indentation.

Reply:
sorry I mean I see them in your code here - the `\t` characters

I think the rest of the indentation in generated code for this project uses spaces instead of tabs, so if we need spacing lets use spaces.

Reply:
Done. I think now it looks fine.